pkg/apis/pipeline/v1alpha1/task_defaults.go

```diff
@@ -26,13 +29,27 @@ func (t *Task) SetDefaults(ctx context.Context) {
 
 // SetDefaults set any defaults for the task spec
 func (ts *TaskSpec) SetDefaults(ctx context.Context) {
+	if v1alpha2.IsUpgradeViaDefaulting(ctx) {
+		fmt.Println("********************************************")
+		v := v1alpha2.TaskSpec{}
+		if ts.ConvertUp(ctx, &v) == nil {
+			fmt.Printf("v: %+v\n", v)
+			fmt.Printf("v.inputs: %+v\n", v.Inputs)
+			fmt.Printf("v.outputs: %+v\n", v.Outputs)
+			alpha1 := TaskSpec{}
+			if alpha1.ConvertDown(ctx, v) == nil {
+				*ts = alpha1
+			}
+		}
+	}
+	fmt.Printf("ts: %+v\n", ts)
```

Comment:
This combined with https://github.com/tektoncd/pipeline/pull/1529/files#diff-58de452513b7e8d8d3cfea23eb4ae6a8R116 make the conversion automatic from `v1alpha1` to a `v1alpha2` compatible object (still stored as `v1alpha1`)